Update 07-fetching-data.mdx (#82862)
vercel/next.js: docs/01-app/01-getting-started/07-fetching-data.mdx

Sequential data fetching happens when nested components in a tree each fetch their own data and the requests are not [deduplicated](/docs/app/guides/caching#request-memoization), leading to longer response times.

<Image

>   alt="Sequential and Concurrent Data Fetching"

Author: Can't change the image file name as that would be destructive, might result in image not displaying due to wrong name.

>   alt="Sequential and Concurrent Data Fetching"

Reviewer: Let's for now keep Parallel here

To improve the user experience, you should use [React `<Suspense>`](/docs/app/getting-started/linking-and-navigating#streaming) to show a `fallback` while data is being fetch. This will enable [streaming](#streaming) and prevent the whole route from being blocked by the sequential data requests.

### Concurrent data fetching

> Concurrent data fetching happens when data requests in a route are eagerly initiated and executed at the same time.

Author: This pull request is majorly to correct the notion on how "Promise" works. Not about the internal workings of how next.js renders the layout and page routes.

> Concurrent data fetching happens when data requests in a route are eagerly initiated and executed at the same time.

Reviewer: Same here, let's revert to where it was

}
```

> You can initiate requests concurrently by defining them outside the components that use the data, and resolving them together, for example, with [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) (or any of the 4 [`Promise` concurrency methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#promise_concurrency), depends on your use case):

Reviewer: Here we can state why these are parallel rather. It is not Promise.all that's the powerful abstraction, but calling fetch('/a') and fetch('/b') right after the other.

```suggestion
Initiate multiple requests by calling fetch and await them together with [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). Requests start immediately when fetch is called. `Promise.all` only waits for all of them to resolve, if any rejects `Promise.all` rejects too. If some failures are acceptable, use [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) to handle successes and errors together.
```

> Start multiple requests by calling `fetch`, then await them with [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). Requests begin as soon as `fetch` is called. `Promise.all` waits for all to resolve, but rejects if any fail. If some failures are acceptable, use [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) to handle successes and errors together.

Reviewer: ```suggestion
Start multiple requests by calling `fetch`, then await them with [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). Requests begin as soon as `fetch` is called.
```

I remembered we do mention Promise.allSettled in a Good To know right after these snippets -

> Initiate multiple requests by calling fetch and await them together with [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). Requests start immediately when fetch is called. `Promise.all` only waits for all of them to resolve, if any rejects `Promise.all` rejects too. If some failures are acceptable, use [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) to handle successes and errors together.

Reviewer: ```suggestion
Start multiple requests by calling `fetch`, then await them with [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). Requests begin as soon as `fetch` is called. `Promise.all` waits for all to resolve, but rejects if any fail. If some failures are acceptable, use [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) to handle successes and errors together.
```

  const artistData = getArtist(username)
  const albumsData = getAlbums(username)

>   // Initiate both requests concurrently

Reviewer: This comment is not at the right level, let's move it to 527 (above getArtist) and just say `initiate requests`

  const artistData = getArtist(username)
  const albumsData = getAlbums(username)

>   // Initiate both requests concurrently

Reviewer: Same as above
